src/zen/split-view/ZenViewSplitter.mjs

```diff
@@ -1894,6 +1903,279 @@ class ZenViewSplitter extends ZenDOMOperatedFeature {
+
+    const URL_TYPES = ['text/uri-list', 'text/x-moz-url', 'text/plain'];
+
+    const FIXUP_FLAGS = Ci.nsIURIFixup.FIXUP_FLAG_FIX_SCHEME_TYPOS;
+
+    const matchedType = URL_TYPES.find((type) => {
+      const raw = dt.getData(type);
+      return typeof raw === 'string' && raw.trim().length > 0;
+    });
+
+    const uriString = dt.getData(matchedType).trim();
+
+    const info = Services.uriFixup.getFixupURIInfo(uriString, FIXUP_FLAGS);
+
+    if (!info || !info.fixedURI) {
+      return null;
+    }
+
+    return info.fixedURI.spec;
+  }
+
+  _handleLinkDragEnter(event) {
+    // If rearrangeViewEnabled - don't do anything
+    if (this.rearrangeViewEnabled) {
+      return;
+    }
+
+    const shouldBeDisabled = !this.canOpenLinkInSplitView();
+    if (shouldBeDisabled) return;
+
+    // If the target is our drop zone or one of its children, or already active, do nothing here.
+    if (
+      this._linkDropZone &&
+      (this._linkDropZone.contains(event.target) || this._linkDropZone.hasAttribute('enabled'))
+    ) {
+      return;
+    }
+
+    // If the data is not a valid URI, we don't want to do anything
+    if (!this._validateURI(event.dataTransfer)) {
+      return;
+    }
+
+    this._isLinkDragging = true;
+    this._showLinkDropZone();
+
+    event.preventDefault();
+    event.stopPropagation();
+  }
+
+  _handleLinkDragLeave(event) {
+    if (
+      event.target === document.documentElement ||
+      (event.clientX <= 0 && event.clientY <= 0) ||
+      event.clientX >= window.innerWidth ||
+      event.clientY >= window.innerHeight
+    ) {
+      if (this._linkDropZone && !this._linkDropZone.contains(event.relatedTarget)) {
+        this._isLinkDragging = false;
+        this._hideLinkDropZone();
+      }
+    }
+  }
+
+  _handleLinkDragDrop(event) {
+    if (!this._linkDropZone || !this._linkDropZone.contains(event.target)) {
+      if (this._linkDropZone && this._linkDropZone.hasAttribute('enabled')) {
+        this._isLinkDragging = false;
+        this._hideLinkDropZone(true); // true for forced hiding
+      }
+    }
+  }
+
+  _handleLinkDragEnd(event) {
+    this._isLinkDragging = false;
+    this._hideLinkDropZone(true); // true for forced hiding
+  }
+
+  _handleDropForSplit(event) {
+    let linkDropZone = this._linkDropZone;
+    event.preventDefault();
+    event.stopPropagation();
+
+    const url = this._validateURI(event.dataTransfer);
+
+    if (!url) {
+      this._hideDropZoneAndResetState();
+      return;
+    }
+
+    const currentTab = gZenGlanceManager.getTabOrGlanceParent(gBrowser.selectedTab);
+    const newTab = this.openAndSwitchToTab(url, { inBackground: false });
+
+    if (!newTab) {
+      this._hideDropZoneAndResetState();
+      return;
+    }
+
+    const linkDropSide = this._calculateDropSide(event, linkDropZone);
+
+    this._createOrUpdateSplitViewWithSide(currentTab, newTab, linkDropSide);
+
+    this._hideDropZoneAndResetState();
+  }
+  _calculateDropSide(event, linkDropZone) {
```

Comment:
This might create some conflicts with already existing functions. Could we move all of these into a separate class / file? Called `ZenSplitViewLinkDrop`